src/Razor/src/Microsoft.VisualStudio.RazorExtension/BindingRedirects.cs

```diff
@@ -1,7 +1,12 @@
 ﻿using Microsoft.VisualStudio.Shell;
 
+
+[assembly: ProvideBindingRedirection(
+    AssemblyName = "Microsoft.VisualStudio.LanguageServer.Protocol.Extensions",
+    CodeBase = @"$PackageFolder$\Microsoft.VisualStudio.LanguageServer.Protocol.Extensions.dll",
+    OldVersionLowerBound = "16.4.30.5595", OldVersionUpperBound = "16.4.30.5595")]
```

Comment:
If you're binding from version X to X (i.e. not rebinding the assembly version at all) is there any benefit (or cost) to using a binding redirect instead of a simpler codebase?  (This question also applies to the other binding redirects.)

Looking at the generated devenv.exe.config as a reference, I see at least 3 other references to different versions of this assembly, but they're all just codebases.

Reply:
This is just my VS noobness shining through and not knowing any better 😄